app/lib/routes/settings.dart

```diff
@@ -71,6 +78,109 @@ class Settings extends StatelessWidget {
   }
 }
 
+class ShareSettings extends StatelessWidget {
+  const ShareSettings({required this.bike, super.key});
+  final Bike bike;
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoScaffold(
+      overlayStyle: SystemUiOverlayStyle.dark,
+      body: Scaffold(
+        appBar: CupertinoNavigationBar(
+          backgroundColor: CupertinoTheme.of(context).scaffoldBackgroundColor,
+          leading: Container(),
+          middle: const Text('Manage sharing'),
+          trailing: _CloseButton(onPressed: () => Navigator.pop(context)),
+        ),
+        body: SafeArea(
+          child: Column(
+            children: [
+              _Section(
+                title: 'Share bike',
+                children: [
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Form(
+                          child: TextFormField(
+                        decoration: const InputDecoration(
+                          labelText: 'Email',
+                        ),
+                        onChanged: (value) {
+                          log(value);
+                        },
+                      )),
+                      ElevatedButton(
+                        onPressed: () {
```

Comment:
Can you disable this button if there is no user input?

Reply:
Done

Reply:
I have tried my best but couldn't figure out a way to do this yet since it is a stateless widget I can't use a state :!

Reply:
It doesn't look disabled, but it doesn't; do anything when you press it now without a correct email.